src/utils/decorators.js

```diff
@@ -0,0 +1,11 @@
+function delayDecorator(callback, dellay) {
+  let timeoutId = null;
+  return (...args) => {
+    clearTimeout(timeoutId);
```

Comment:
do you need to check here that timeoutId is not null before making clearTimeout ?

Reply:
Done. I have added a condition.